index.js

```diff
@@ -1,32 +1,52 @@
-const client = require('./lib/statsd');
+/*
+ * Initialize a hot-shots client.
+ */
 
-const isNotEmpty = (obj) => obj && Object.getOwnPropertyNames(obj).length;
+const os = require('os');
 
-function keyValuesToString(keyValues) {
-  return Object.keys(keyValues)
-    .reduce((r, k) => r.concat(`${k}=${keyValues[k]}`), [])
-    .join(',');
+const statsdClient = require('hot-shots');
+const strigoLogger = require('strigo-node-logger');
+
+const STATSD_HOST = '127.0.0.1';
+const STATSD_PORT = 8125;
+const CONTAINER_HOSTNAME = process.env.HOSTNAME;
+const NOMAD_ALLOC_ID = process.env.NOMAD_ALLOC_ID;
+
+
+function setupStatsdClient({ svc, env, host = STATSD_HOST, port = STATSD_PORT }) {
+  const log = strigoLogger.setupNodeLogger({ env, level: 'info' });
+  if (!svc) {
+    throw new Error('Must provide a service name');
+  }
+
+  const globalTags = { env };
+
+  // If we don't add these conditionals, these tags will be set with an `undefined` value, which is wasteful.
+  if (CONTAINER_HOSTNAME) {
+    globalTags.container_hostname = CONTAINER_HOSTNAME;
+  }
+  if (NOMAD_ALLOC_ID) {
+    globalTags.nomad_alloc_id = NOMAD_ALLOD_ID;
+  }
+
+  log.info(`Initializing statsd client for service ${svc} @ ${STATSD_HOST}:${STATSD_PORT}...`);
+  const client = new statsdClient.StatsD(
+    {
+      host,
+      port,
+      prefix: `${svc}_`,
+      globalTags,
+      telegraf: true,
+    },
+  );
+
+  client.close(function(err) {
```

Comment:
https://github.com/brightcove/hot-shots#check - at the end of the example it states that it closes the client, but it could take time then there is a callback function...

Reply:
My bad.